Mathlib/Order/Category/NonemptyFinLinOrd.lean

```diff
@@ -235,3 +235,8 @@ def nonemptyFinLinOrdDualCompForgetToFinPartOrd :
       forget₂ NonemptyFinLinOrd FinPartOrd ⋙ FinPartOrd.dual where
   hom := { app := fun X => OrderHom.id }
   inv := { app := fun X => OrderHom.id }
+
+theorem Fin.le_succ {n} (i : Fin n) : i.castSucc ≤ i.succ := Nat.le_succ i
```

Comment:
Belongs in some file about `Fin`?

Reply:
@semorrison I couldn't for the life of me figure out where the category structure on Fin is discussed. I agree, this doesn't feel like the most natural location.

Reply:
But this lemma has nothing to do with the category structure. It is just an inequality, so should go in Mathlib.Data.Fin.Basic (or even earlier; I'm happy to move this to the core repository). It should also be renamed `Fin.castSucc_le_succ` as I think it's ambiguous otherwise.

Reply:
Ah sorry. Now I understand the point you were making. Yes please move it wherever it belongs and rename appropriately.

Reply:
@kim-em this was fixed by @digama0